subprojects/packagefiles/speexdsp/meson.build

```diff
@@ -0,0 +1,112 @@
+project('speexdsp',
+        'c',
+        version : '1.2.0',
+        license : 'BSD-3-Clause',
+        default_options : [
+          'c_std=c99',
+          'default_library=both',
+          'warning_level=0'])
+
+# library versioning from speexdsp-1.2.0's configure.ac
+# SPEEXDSP_LT_CURRENT=6, SPEEXDSP_LT_REVISION=1, SPEEXDSP_LT_AGE=5
+lib_current = 6
+lib_revision = 1
+lib_age = 5
+lib_soversion = lib_current - lib_age
+lib_version = '@0@.@1@.@2@'.format(lib_soversion, lib_age, lib_revision)
+
+c_args = []
+cc = meson.get_compiler('c')
+host_cpu_family = host_machine.cpu_family()
+
+# Round up configuration data
+add_project_arguments('-DHAVE_CONFIG_H',
+                      language: 'c')
+cdata = configuration_data()
+
+# Set SIMD config data and flags
```

Comment:
Interestingly, upstream has --enable-sse=[yes|no] and --enable-neon=[yes|no] options. If not passed on the command line, one is automatically selected based on the current system. (Basically a Meson combo with a "check" option.)

The default behavior is to check if it can be enabled, but it's possible to manually configure this off and generate really portable libraries that support being moved off the current machine and onto one that uh doesn't even support sse2 because that's incredibly realistic, right?

Reply:
> upstream has --enable-sse=[yes|no] and --enable-neon=[yes|no] 

I went with a default-enabled bool, called "simd", because it's already architecture-specific at the compiler level.  In the rare case someone wants to ship binaries for pentium3 and lesser hardware, they can toggle this off.

Reply:
Yup, makes perfect sense.

Having separate options seems rather pointless, and the autotools default is to enable it anyway (via checking) so your implementation seems very sane.